### DISCUSSION TITLE
Failed download handling

### DISCUSSION THREAD

**Discussion by a-battenberg** [aclap-dev/video-downloadhelper#1480] [Help and Q&A]
### Your question

Hi, I love the software and use it for a long time now.
Recently I discovered, that the listing of a video get just removed after the download failed. I just get a notification on top of my list that a download failed, but it does not show which one failed. This is the case for me in Firefox as well as Chrome (both with a license). 

If I remember correctly, in the past, the failed video/download got marked with a red banner. 
Now it is frustrating to check my download folder for the one file that is a failed download.

It would be nice if the failed download would just be highlighted instead of just disappearing, since the incomplete file is still in the download folder.

### Addon & CoApp version and details

Microsoft Windows 10 Home
Version 22H2
build 19045.5131

version: 9.1.0.44
target: mozilla
channel: stable
lang: de
coapp: {"found":true,"path":"C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe","version":"2.0.19","new_version":false}
license: {"accepted":true,"email":"<removed>","key":"<removed>"}
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:132.0) Gecko/20100101 Firefox/132.0
{
    "valid_license_message_has_been_displayed": true,
    "used_history_button": true,
    "download_directory": "<removed>",
    "concurrent_downloads_max": 20,
    "theme": "dark",
    "view_options": {},
    "use_legacy_ui": false,
    "use_sidebar": true,
    "open_count_store": 3,
    "successfull_dl": 73,
    "download_history": {
        "__serializer_tag": "map",
        "__serializer_value": []
    }
}

version: 9.1.1.0
target: google
channel: stable
lang: de
coapp: {"found":true,"new_version":false,"path":"C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe","version":"2.0.19"}
license: {"accepted":true,"email":"<removed>","key":"<removed>"}
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/130.0.0.0 Safari/537.36
{
    "valid_license_message_has_been_displayed": true,
    "used_history_button": true,
    "download_directory": "<removed>",
    "never_show_no_incognito_msg_again": true,
    "view_options": {
        "all_tabs": false,
        "hide_downloaded": false,
        "low_quality": false,
        "show_button_clean": true,
        "show_button_clean_all": false,
        "show_button_convert_local": false,
        "sort_by_status": true,
        "sort_reverse": false
    },
    "use_sidebar": true,
    "open_count_store": 8,
    "successfull_dl": 18,
    "never_show_successfull_dl_message": true,
    
    "record_download_history": true,
}

**Comment by GrampaWildWilly** [comment]
I do not work for VDH.  I'm just another user.

Your information says, & I quote:

> "record_download_history": true,

In that case, you ought to look in your VDH history & see what's there.  It might help you out.  But that appears to be only for your Chrome installation.  You don't appear to have enabled the history in your Firefox.  You probably ought to.

**Reply by a-battenberg** [reply]
I tested it now with the history activated in Firefox.
After a download failed, is got the message that something went wrong, but it showed not which download it was. The downlaod history did not show the failed attempt, it just got deleted (it was one episode of a series with the episode number in the title).

**Reply by GrampaWildWilly** [reply]
That's too bad.  Oh well.  You should do what I do.  Whenever I do a download, I always leave the browser window/tab open until I'm sure that download has completed successfully.  I also use `Download As`, never `Download`.  That way, I can assign file names that make sense to me.  If something fails, I can tell because the file with the relevant name won't exist.  I can just go back to the same browser tab & retry the download or try some other evasive maneuver.

Should VDH do this better & make life easier for us?  Probably.  But it doesn't.  So adapt to reality.

**Reply by diox8tony** [reply]
I have the same issue. 

"Failed to download media",,,is an empty error, that gives you no ability to re-download the media. 
It doesn't show you the link for the webpage, for you to manually re-start the DL.

This sucks when you have a queue of 100 files, when you come back, 80 have finished and 20 are errors. 
Which 20 failed? how am I going to find out which files I need to download?

The error should auto-retry(setting for this) + It should include a button to "retry", + ALL items in the queue (waiting, finished, and errors) should all have the link back to the webpage for that DL, that users can click on.
